user/tests/integration/wiring/no_fixture_i2c/i2c.cpp

```diff
@@ -0,0 +1,152 @@
+    // Tracker platforms should have buffers at least 512 bytes large, allocated by the system on startup
+    // IE trying to re-initialize the HAL with same size should fail
+    hal_i2c_config_t smallBuffers = allocateWireConfig(TRACKER_MINIMUM_BUFFER_SIZE);
+    hal_i2c_interface_t largeBufferInterface = (PLATFORM_ID == PLATFORM_TRACKER) ? HAL_I2C_INTERFACE2 : HAL_I2C_INTERFACE1;
+    assertEqual(hal_i2c_init(largeBufferInterface, &smallBuffers), (int)SYSTEM_ERROR_NOT_ENOUGH_DATA);
+    freeWireBuffers(&smallBuffers);
+#endif
+
+}
+
+test(I2C_01_test_acquire_wire_buffer) {
+    // Construct TwoWire object with user acquireWireBuffer() configuration structures, which should re-initialize the I2C HAL
+    Wire.begin();
+
+    // Attempt to re-initialize the HAL with different sized buffers to implicitly determine if the user buffers were used by the HAL
+    // System Defaults < User values from acquireWireBuffer < Values used here to re-initialize the HAL
+
+    // Equal size buffers should fail
+    hal_i2c_config_t wireSmallerBuffers = allocateWireConfig(WIRE_ACQUIRE_BUFFER_SIZE);
+    assertEqual(hal_i2c_init(HAL_I2C_INTERFACE1, &wireSmallerBuffers), (int)SYSTEM_ERROR_NOT_ENOUGH_DATA);
+    freeWireBuffers(&wireSmallerBuffers);
+
+    // Larger sized buffers should succeed
+    hal_i2c_config_t wireLargerBuffers = allocateWireConfig(WIRE_ACQUIRE_BUFFER_SIZE + 1);
+    assertEqual(hal_i2c_init(HAL_I2C_INTERFACE1, &wireLargerBuffers), (int)SYSTEM_ERROR_NONE);
+    freeWireBuffers(&wireLargerBuffers);
+
+    // Reinitializing with a NULL config is not allowed
+    assertEqual(hal_i2c_init(HAL_I2C_INTERFACE1, nullptr), (int)SYSTEM_ERROR_INVALID_ARGUMENT);
+
+#if Wiring_Wire1
+    Wire1.begin();
+    hal_i2c_config_t wire1SmallerBuffers = allocateWireConfig(WIRE1_ACQUIRE_BUFFER_SIZE);
+    assertEqual(hal_i2c_init(HAL_I2C_INTERFACE2, &wire1SmallerBuffers), (int)SYSTEM_ERROR_NOT_ENOUGH_DATA);
+    freeWireBuffers(&wire1SmallerBuffers);
+
+    hal_i2c_config_t wire1LargerBuffers = allocateWireConfig(WIRE1_ACQUIRE_BUFFER_SIZE + 1);
+    assertEqual(hal_i2c_init(HAL_I2C_INTERFACE2, &wire1LargerBuffers), (int)SYSTEM_ERROR_NONE);
+    freeWireBuffers(&wire1LargerBuffers);
+
+    assertEqual(hal_i2c_init(HAL_I2C_INTERFACE2, nullptr), (int)SYSTEM_ERROR_INVALID_ARGUMENT);
+#endif
+}
+
+#if HAL_PLATFORM_EXTERNAL_RTC
+
+test(I2C_02_long_read)
+{
+    // Test > 32 byte reads by reading from AM18x5 RTC peripheral
+    Wire1.begin();
+
+    // Read full register range
+    const uint8_t REGISTER_RANGE = 64;
+    uint8_t longReadBuffer[REGISTER_RANGE] = {};
+    uint8_t readLength = REGISTER_RANGE; 
+
+    // Check read values against ID registers
+    const uint8_t ID0_REGISTER = 0x28;
+    const uint8_t ID0_REGISTER_VALUE = 0x18;
+    const uint8_t ID1_REGISTER = 0x29;
+    const uint8_t ID1_REGISTER_VALUE = 0x05;
+
+    hal_i2c_interface_t wireInterface = HAL_PLATFORM_EXTERNAL_RTC_I2C;
```

Comment:
It looks like only Tracker platform has the external RTC (not trackerm). If there is another platform that does incorporate an RTC, it just needs to define `HAL_PLATFORM_EXTERNAL_RTC`, `HAL_PLATFORM_EXTERNAL_RTC_I2C`, `HAL_PLATFORM_EXTERNAL_RTC_I2C_ADDR` and this test should work.

Reply:
What about using PMIC or FG?

Reply:
The MAX17043 fuel gauge only has [15 bytes of register data](https://datasheets.maximintegrated.com/en/ds/MAX17043-MAX17044.pdf)
The BQ24195 PMIC only has [10 bytes of register data](https://www.ti.com/lit/ds/symlink/bq24195.pdf?ts=1666034756902&ref_url=https%253A%252F%252Fwww.ti.com%252Fproduct%252FBQ24195)
So only the RTC has >32 bytes of contiguous register data to conveniently read (or the actual Quectel GNSS device on trackerm, which I dont have available to test against)

Reply:
I think most of the times you just wrap around at the maximum. Might be worth trying, all we need for the IC to not NACK

Reply:
I can indeed read more bytes than the device has available from the BQ24195 PMIC:
```
0000002665 [app] INFO: I2C_03_long_read_2 result 64                                                                                                                                                                                                                                                                                   
021b18119a8a034b6c0023ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff
```
I will keep the RTC test and add one for the PMIC? Is it ok to have both, or are they too similar/redundant?

Reply:
I'd say add both. Make sure to enable PMIC only for platforms with `HAL_PLATFORM_POWER_MANAGEMENT_OPTIONAL == 0`

Reply:
Ok, checked for `HAL_PLATFORM_POWER_MANAGEMENT_OPTIONAL == 0`, which limits this to tracker it seems.